From 79b96cd5ed49095ee2b2d7219b5dfa3746505458 Mon Sep 17 00:00:00 2001
From: Guenter Kukkukk <kukks@samba.org>
Date: Tue, 1 Jul 2014 17:43:55 +0200
Subject: [PATCH 2/5] mount.cifs: on 2nd try mount.cifs must also uppercase
 "orig_dev"

Recent kernels now ignore "unc=..." mount option. mount.cifs, when
getting errno=ENXIO, retries the mount with uppercased hostname,
sharename and prefixpath in the "unc=..." mount option, which is ignored
now in the kernel. Used e.g. during OS/2 mounts, which fail now.

Also uppercase the now used "orig_dev" parameter.

Signed-off-by: Guenter Kukkukk <kukks@samba.org>
(cherry picked from commit 43fd65ba0fca85a86a79e84bb2bc42b531d858e6)
Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---
 mount.cifs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mount.cifs.c b/mount.cifs.c
index 497665d..3535096 100644
--- a/mount.cifs.c
+++ b/mount.cifs.c
@@ -2097,7 +2097,8 @@ mount_retry:
 			if (!already_uppercased &&
 			    uppercase_string(parsed_info->host) &&
 			    uppercase_string(parsed_info->share) &&
-			    uppercase_string(parsed_info->prefix)) {
+			    uppercase_string(parsed_info->prefix) &&
+			    uppercase_string(orig_dev)) {
 				fprintf(stderr,
 					"Retrying with upper case share name\n");
 				already_uppercased = 1;
-- 
2.5.5

